Source/Core/Core/Scripting/Python/Pyr.h

```diff
@@ -0,0 +1,203 @@
+#pragma once
+
+#include <Python.h>
+
+namespace Pyr
+{
+class Object
+{
+  PyObject* m_py_object_;
+  Object(PyObject* py_object) : m_py_object_(py_object) {}
+
+public:
+  Object& operator=(Object&&) = default;
+  Object& operator=(const Object& rhs)
+  {
+    Py_XINCREF(rhs.Unwrap());
+    Py_XDECREF(m_py_object_);
+    m_py_object_ = rhs.Unwrap();
+    return *this;
+  }
+  Object(Object&&) = default;
+  Object(Object& rhs)
+  {
+    m_py_object_ = rhs.Unwrap();
+    Py_XINCREF(rhs.Unwrap());
+  }
+
+  ~Object() { Py_XDECREF(m_py_object_); }
+  PyObject* Unwrap() const { return m_py_object_; }
+  bool IsNull() const { return m_py_object_ == NULL; }
+  static Object WrapExisting(PyObject* py_object) { return Object(py_object); }
+};
+
+inline Object Wrap(PyObject* py_object)
+{
+  return Object::WrapExisting(py_object);
+}
+inline Object Take(PyObject* py_object)
+{
+  Py_XINCREF(py_object);
+  return Object::WrapExisting(py_object);
+}
+
+inline Object& Null()
+{
+  static Object null_obj = Wrap(nullptr);
+  return null_obj;
+}
+
+template <typename T>
+constexpr PyObject* ConvertToPy(T n)
+{
+  // using the constexpr checks is easier than template overloading imo
+  if constexpr (std::is_same<T, bool>())
+  {
+    return PyBool_FromLong(n);
+  }
+  else if constexpr (std::is_floating_point<T>::value)
```

Comment:
Yeah, I didn't forgot about those, don't worry. I just didn't get around implementing all your suggestions yet. Since I'm using C++17 right now, I could also use `std::is_floating_point_v<T>` and the other `*_v` variants, if that's preferable.